packages/kit/src/hooks/useFlowRevertibleRandom.ts

```diff
@@ -0,0 +1,53 @@
+import {UseQueryOptions, UseQueryResult} from "@tanstack/react-query"
+import {useFlowQuery} from "./useFlowQuery"
+
+export interface RevertibleRandomResult {
+  blockHeight: string
+  value: string
+}
+
+export interface UseFlowRevertibleRandomArgs {
+  min?: string
+  max: string
+  query?: Omit<UseQueryOptions<unknown, Error>, "queryKey" | "queryFn">
+}
+
+export function useFlowRevertibleRandom({
+  min = "0",
+  max,
+  query: queryOptions = {},
+}: UseFlowRevertibleRandomArgs): UseQueryResult<RevertibleRandomResult, Error> {
+  const mergedQuery = {
+    staleTime: Infinity,
+    refetchOnWindowFocus: false,
+    ...queryOptions,
+  }
+
+  const result = useFlowQuery({
+    cadence: `
+      access(all) struct RevertibleRandomResult {
+        access(all) let blockHeight: UInt64
+        access(all) let value: UInt256
+
+        init(blockHeight: UInt64, value: UInt256) {
+          self.blockHeight = blockHeight
+          self.value = value
+        }
+      }
+
+      access(all) fun main(min: UInt256, max: UInt256): RevertibleRandomResult {
```

Comment:
I'm wondering now if we might want to allow for a count of random values. Under the hood, there's a PRG that's used for a `revertibleRandom` call on any given block. So if I spam this script, I'll get the same random result for each `useFlowRandom` call that pings the same block. But this behavior may not be intuitive to someone thinking they'll get a different random value for every `useFlowRandom` call.
However, if instead we expose the ability to get several random values for every `useFlowRandom` call, the PRG state advances in the script context thus allowing for `n` unique random values that are generated in the same script even though they're generated in the same block.
This would be an interface change to the hook, so raising this for consideration. Thoughts?